asapdiscovery-data/asapdiscovery/data/openeye.py

```diff
@@ -728,7 +804,7 @@ def _set_SD_data_repr(mol: oechem.OEMol, data: dict[str, Any]) -> oechem.OEMol:
         OpenEye OEMol with SD data set
     """
     # NOTE: use repr to ensure re-reading the SD data will give the same value
-    mol = set_SD_data(mol, {k: repr(v) for k, v in data.items()})
```

Comment:
Change comment above if changing this, might also want to wrap in try/except for string cast, unless guaranteed that all incoming objects will be castable.

Reply:
Good catch I think everything from the ligand side should be a string as we are enforcing it via typing and using json models so we should be safe to change the type hint to expect dicts of str only.